src/main/java/org/jpeek/calculus/GeneralCalculation.java

```diff
@@ -0,0 +1,181 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.jpeek.calculus;
+
+import com.amazonaws.util.IOUtils;
+import com.jcabi.log.Logger;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Calculates aggregated value for cohesion metrics.
+ *
+ * @since 1.0.0
+ * @checkstyle StringLiteralsConcatenationCheck (250 lines)
+ */
+public class GeneralCalculation {
+
+    /**
+     * Created report with aggregated value.
+     *
+     * @param target Path to report file
+     * @param metrics List of metrics used in calculation
+     * @throws IOException If fails
+     */
+    @SuppressWarnings({
+        "PMD.AvoidDuplicateLiterals",
+        "PMD.ExcessiveMethodLength"
+    })
+    public void createReport(final Path target, final List<String> metrics) throws IOException {
+        final Map<String, MetricPresentation> res = new HashMap<>();
+        for (final String metric : metrics) {
+            res.put(
+                metric,
+                new MetricPresentation(
+                    String.format(
+                        "%s/%s.html",
+                        target.toString(),
+                        metric
+                    )
+                )
+            );
+        }
+        final Map<String, String[]> output = getResult(res);
+        String template;
+        template = IOUtils.toString(
+            GeneralCalculation.class.getResourceAsStream("general_calculus.html")
+        );
+        final String classesout = String.format(
+            "<p id=\"classes\">Classes: %s.</p>\n",
+            output.keySet().size()
+        );
+        String tbody = "";
+        for (final Map.Entry<String, String[]> entry : output.entrySet()) {
+            tbody = tbody.concat(
+                String.format(
+                    "%s%s%s%s%s%s", "<tr>\n",
+                    String.format(
+                        "               <td><code title=\"%s\">%s</code></td>\n",
```

Comment:
@iMaks99 the code here is hard to understand because you mix data and formatting in one place. XSL is used exactly in order to solve this problem: the data stays in XML, while the formatting is in XSL.